app/controllers/vm_cloud_controller.rb

```diff
@@ -21,7 +21,11 @@ def resize
       ) unless @explorer
       @flavors = {}
       unless @record.ext_management_system.nil?
-        @record.ext_management_system.flavors.each { |f| @flavors[f.name] = f.id unless f == @record.flavor }
+        @record.ext_management_system.flavors.each do |f|
+          unless f == @record.flavor || @record.flavor.root_disk_size > f.root_disk_size
```

Comment:
how about renaming `f` to `ems_flavor` or `provider_flavor` or something similar for clarity?